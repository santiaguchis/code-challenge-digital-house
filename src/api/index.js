export * from './apiRest';
export * from './services';