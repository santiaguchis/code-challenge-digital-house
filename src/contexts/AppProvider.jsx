import { useState } from 'react';
import { AppContext } from './AppContext';

const initialState = {
  user: {
    firstName: 'Ruben',
    lastName: 'Rodriguez'
  }
}

export const AppProvider = ({ children }) => {
  const [ movements, setMovements ] = useState([]);
  const [ filtered, setFiltered ] = useState([]);
  const [ totalPoints, setTotalPoints ] = useState(0);
  const [ filterSelected, setFilterSelected ] = useState('');

  const [ isLoading, setIsLoading ] = useState(true);

  const onHandleFilter = (items) => {
    setIsLoading(false);
    setFiltered(items)
  }
  const appState = {
    ...initialState,
    isLoading,
    movements,
    filtered,
    totalPoints,
    filterSelected,
    setMovements,
    setFiltered,
    setTotalPoints,
    setFilterSelected,
    setIsLoading,
    onHandleFilter,
  }
  return (
    <AppContext.Provider
      value={appState}
    >
      { children }
    </AppContext.Provider>
  )
}