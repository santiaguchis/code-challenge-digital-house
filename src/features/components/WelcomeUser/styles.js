import { StyleSheet, StatusBar } from 'react-native';

import { theme } from 'src/core/utils/styles';

const styles = StyleSheet.create({
  container: {
    marginTop: StatusBar.currentHeight,
  },
  title: {
    color: theme.colors.primaryText,
    fontSize: 20,
    fontWeight: '800',
    lineHeight: 27.32
  },
  subTitle: {
    color: theme.colors.primaryText,
    fontSize: 16,
    fontWeight: '400',
    lineHeight: 21.86
  }
})

export default styles;