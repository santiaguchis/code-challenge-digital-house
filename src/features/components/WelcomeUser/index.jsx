import { useContext } from 'react';
import { Text, View } from 'react-native'
import { AppContext } from 'src/contexts';

import styles from './styles';

const {
  title,
  subTitle,
  container,
} = styles;

const WelcomeUser = () => {
  const store = useContext( AppContext );
  const {
    firstName,
    lastName
  } = store.user
  return (
    <View style={ container }>
      <Text style={ title }>Bienvenido de vuelta!</Text>
      <Text style={ subTitle }>{firstName} {lastName}</Text>
    </View>
  )
}

export default WelcomeUser;
