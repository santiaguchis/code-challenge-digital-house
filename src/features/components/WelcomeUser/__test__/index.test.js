import { render, screen } from '@testing-library/react-native';
import { AppProvider } from 'src/contexts';

import WelcomeUser from 'src/features/components/WelcomeUser';

let component;
describe('<WelcomeUser /> features/component', () => {
  const findTextUsername = 'Ruben Rodriguez';
  beforeEach(() => {
    component = render(
      <AppProvider>
        <WelcomeUser />
      </AppProvider>
    )
  })

  it('renderer element', () => {
    expect(component).toBeDefined();
  })

  it('Check text in component', () => {
    const textInComponent = screen.getByText(findTextUsername);
    expect(textInComponent).toHaveTextContent(findTextUsername);
  })

  it('Check snapshot', () => {
    expect(screen.toJSON()).toMatchSnapshot();
  })
})