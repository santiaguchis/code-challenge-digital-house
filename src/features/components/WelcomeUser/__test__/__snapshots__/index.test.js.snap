// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WelcomeUser /> features/component Check snapshot 1`] = `
<View
  style={{}}
>
  <Text
    style={
      {
        "color": "#000000",
        "fontSize": 20,
        "fontWeight": "800",
        "lineHeight": 27.32,
      }
    }
  >
    Bienvenido de vuelta!
  </Text>
  <Text
    style={
      {
        "color": "#000000",
        "fontSize": 16,
        "fontWeight": "400",
        "lineHeight": 21.86,
      }
    }
  >
    Ruben
     
    Rodriguez
  </Text>
</View>
`;
